app/assets/javascripts/rentals/new.js.erb

```diff
@@ -55,10 +54,53 @@ function costCalculation(){
    }
 }
 
+function userResultHandleClick(e) {
+  var userId = e.currentTarget.id.substr(10); //select the user id
+  $("#rental_user_id").val(userId); 
+  
+  var row = $(e.currentTarget.parentNode.parentNode); //select the row
+  var name = row.find("td:nth-child(1)").text() + " " + row.find("td:nth-child(2)").text(); //first two td's joined by space
+  $("#rental_userSearchQuery").val(name); //set the user to the full name
+
+  $("#modalSearchResults").modal('toggle'); //close modal
+}
+
+function ajaxUserSearch(pageNo) {
+  var query = $("#rental_userSearchQuery").val();
+
+  //inject page into request
+  var params = $.extend({user_search_query: query}, pageNo ? {page: pageNo} : {page: "1"} );
+
+  $.ajax({
+    url: Routes.rentals_search_users_path(),
+    dataType: "html",
+    data: params,
+    success: function (data) {
+      $("#userResults").html(data); //put content in the modal
+      
+      //attach event listeners to results
+      $(".userResult").click(userResultHandleClick);
+
+      //need to reformat links
+      $("#userResults .pagination a").each(modalSearchResultsFormatPaginationLinks);
+    },
+    error: function () {
+      alert("Failed to retreive users.");
+    }
+  });
+}
+
+function modalSearchResultsFormatPaginationLinks(undefined, link) {
+  link = $(link); //jquery that
+  var pageNo = link.attr("href").split("&")[0].split("?")[1].split("=")[1]; //works universially to pull pageNo out of link
```

Comment:
unfortunately this is a very specific thing, using will_paginate within a partial retrieved by an ajax request.  I searched but was unable to find an existing library.